src/app/ReadClient.cpp

```diff
@@ -296,6 +308,28 @@ CHIP_ERROR ReadClient::GenerateAttributePathList(AttributePathIBs::Builder & aAt
     return aAttributePathIBsBuilder.GetError();
 }
 
+CHIP_ERROR ReadClient::GenerateDataVersionFilterList(DataVersionFilterIBs::Builder & aDataVersionFilterIBsBuilder,
+                                                     DataVersionFilterParams * apDataVersionFilterParamsList,
+                                                     size_t aDataVersionFilterParamsListSize)
+{
+    for (size_t index = 0; index < aDataVersionFilterParamsListSize; index++)
+    {
+        VerifyOrReturnError(apDataVersionFilterParamsList[index].IsValidDataVersionFilter(),
+                            CHIP_ERROR_IM_MALFORMED_ATTRIBUTE_PATH);
+        DataVersionFilterIB::Builder & filter = aDataVersionFilterIBsBuilder.CreateDataVersionFilter();
+        ReturnErrorOnFailure(aDataVersionFilterIBsBuilder.GetError());
+        ClusterPathIB::Builder & path = filter.CreatePath();
+        ReturnErrorOnFailure(filter.GetError());
+        ReturnErrorOnFailure(path.Endpoint(apDataVersionFilterParamsList[index].mEndpointId)
+                                 .Cluster(apDataVersionFilterParamsList[index].mClusterId)
+                                 .EndOfClusterPathIB()
+                                 .GetError());
```

Comment:
OK, but where are we encoding the actual dataversion?

Reply:
good catch, just fix it, thanks